src/DataProvider.js

```diff
@@ -72,6 +72,23 @@ export default class DataProvider {
     }, this.polling())
   }
 
+  getDataWithRetry() {
+    return new Promise((resolve, reject) => {
+      let lastTimeout
+      let timedGetData = async (retries) => {
+        if (retries <= 0) {
```

Comment:
There should probably be a `<` instead.

Reply:
Yes, you're right. E.g. if I wanted 0 retries, `getData` should still be called once...